test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -105,11 +107,14 @@ public void stop() {
             }
         }
 
-        for (QuarkusTestResourceLifecycleManager i : ServiceLoader.load(QuarkusTestResourceLifecycleManager.class)) {
-            testResourceRunners.add(i);
+        Collections.sort(testResourceRunners, new QuarkusTestResourceLifecycleManagerComparator());
+
+        for (QuarkusTestResourceLifecycleManager quarkusTestResourceLifecycleManager : ServiceLoader
+                .load(QuarkusTestResourceLifecycleManager.class)) {
+            testResourceRunners.add(quarkusTestResourceLifecycleManager);
         }
 
-        return testResourceRunners;
+        return new HashSet<>(testResourceRunners);
```

Comment:
You will lose the ordering with this call. I think this method should return a `List` not a `Set`.